araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,346 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
+        which_distance : str
+            The distance of the reconstruction for analysis, following the 
+            convention of other functions in this class. 
+        return_csw : bool
+            If True, will return the CSW_SNR and a ROOT.TGraph of the CSW
+
+        Returns
+        -------
+        csw : ROOT.TGraph
+            The coherently summed waveform. Only returned if `return_csw==True`
+        """
+
+        # Check that reco_results have been passed in properly
+        if 'theta' not in reco_results.keys():
+            raise ValueError(
+                "reco_result object passed into this function does not appear "
+                "to be for a specific reconstruction. Should have a 'theta' key "
+                "but only has the following:", reco_results.keys())
+        
+        # Load wavepacket with waveforms
+        wavepacket= data.get_wavepacket(
+            useful_event=useful_event, which_traces='filtered')
+
+        # Determine which channels should be used in the csw
+        if excluded_channels is not None: 
+            if not isinstance(excluded_channels, np.ndarray):
+                excluded_channels = np.array(excluded_channels)
+        else: 
+            excluded_channels = np.array([])
+        channels_to_csw = []
+        for ch_ID in data.rf_channel_indices: 
+            if ((ch_ID not in excluded_channels) and 
+                (data.rf_channel_polarizations[ch_ID] == polarization)):
+                channels_to_csw.append(ch_ID)
+
+        # In case some channels have different lengths than others, choose the
+        #   smallest waveform size for the length of the csw
+        csw_length = 123456
+        for ch_ID in channels_to_csw:
+            if wavepacket['waveforms'][ch_ID].GetN() < csw_length: 
+                csw_length = wavepacket['waveforms'][ch_ID].GetN() 
+
+        # Determine the "reference channel" to base the CSW around as the
+        #   channel with the maximum voltage
+        max_voltage = {'voltage': -1, 'channel':-123456}
+        for ch_ID in channels_to_csw:
+            this_max_voltage = np.max(wavepacket['waveforms'][ch_ID].GetY())
+            if this_max_voltage > max_voltage['voltage']:
+                max_voltage['voltage'] = this_max_voltage
+                max_voltage['channel'] = ch_ID 
+        reference_ch = max_voltage['channel']
+
+        # Get arrival delays relative to the reference channel based on
+        #   expected arrival times from reconstruction results
+        arrival_delays_reco = self.__get_arrival_delays_reco(
+            data, reco_results, channels_to_csw, reference_ch, 
+            which_distance, solution)
+        
+        # Get arrival delays by zooming in on the cross correlation between
+        #   each channel and the reference channel around the expected
+        #   arrival delay from reconstruction results
+        if useful_event.isSoftwareTrigger(): wf_type = 'soft'
```

Comment:
Sorry can you make a line break so these if/else statements are more readable? (Clearly this is a petpeeve of mine lol)